tasks/lint.rake

```diff
@@ -0,0 +1,69 @@
+require "open3"
+
+#
+# Common stuff for a linter
+#
+module LinterMixin
+  def run
+    offenses = []
+
+    applicable_files.each do |file|
+      if clean?(file)
+        print "."
+      else
+        offenses << file
+        print "F"
+      end
+    end
+
+    print "\n"
+
+    return if offenses.empty?
+
+    raise failure_message_for(offenses)
+  end
+
+  private
+
+  def applicable_files
+    Open3.capture2("git grep -Il ''")[0].split
+  end
+
+  def failure_message_for(offenses)
+    msg = "#{self.class.name} detected offenses. "
+
+    msg += if respond_to?(:fixing_cmd)
+             "Run `#{fixing_cmd(offenses)}` to fix them."
+           else
+             "Affected files: #{offenses.join(' ')}"
+           end
+
+    msg
+  end
+end
+
+#
+# Checks trailing new lines in files
+#
+class MissingTrailingCarriageReturn
```

Comment:
why you don't use rubocop's `Layout/TrailingBlankLines` for that?

Reply:
I use this check across my projects to ensure all files on a project ensure have a trailing new line character. This avoids git warnings, for example.

Rubocop can help for ruby files, but I like to enforce it for all kinds of files.